client/python/kserve-api/samples/http_async_infer_resnet.py

```diff
@@ -0,0 +1,191 @@
+        ssl_options=ssl_options,
+        verbose=False)
+
+    processing_times = np.zeros((0), int)
+
+    # optional preprocessing depending on the model
+    imgs = np.load(args['images_numpy_path'],
+                   mmap_mode='r', allow_pickle=False)
+    imgs = imgs - np.min(imgs)  # Normalization 0-255
+    imgs = imgs / np.ptp(imgs) * 255  # Normalization 0-255
+    # imgs = imgs[:,:,:,::-1] # RGB to BGR
+    print('Image data range:', np.amin(imgs), ':', np.amax(imgs))
+    # optional preprocessing depending on the model
+
+    if args.get('labels_numpy_path') is not None:
+        lbs = np.load(args['labels_numpy_path'],
+                      mmap_mode='r', allow_pickle=False)
+        matched_count = 0
+        total_executed = 0
+    batch_size = int(args.get('batchsize'))
+
+    while batch_size >= imgs.shape[0]:
+        imgs = np.append(imgs, imgs, axis=0)
+        if args.get('labels_numpy_path') is not None:
+            lbs = np.append(lbs, lbs, axis=0)
+
+    iterations = int((imgs.shape[0]//batch_size) if not (args.get(
+        'iterations') or args.get('iterations') != 0) else args.get('iterations'))
+
+    print('Start processing:')
+    print('\tModel name: {}'.format(args.get('pipeline_name') if bool(
+        args.get('pipeline_name')) else args.get('model_name')))
+    print('\tIterations: {}'.format(iterations))
+    print('\tImages numpy path: {}'.format(args.get('images_numpy_path')))
+    if args.get('transpose_input') == "True":
+        if args.get('transpose_method') == "nhwc2nchw":
+            imgs = imgs.transpose((0, 3, 1, 2))
+        if args.get('transpose_method') == "nchw2nhwc":
+            imgs = imgs.transpose((0, 2, 3, 1))
+    print('\tNumpy file shape: {}\n'.format(imgs.shape))
+
+    iteration = 0
+    is_pipeline_request = bool(args.get('pipeline_name'))
+
+    while iteration < iterations:
+        data = []
+        lb = []
+        for x in range(0, imgs.shape[0] - batch_size + 1, batch_size):
+            iteration += 1
+            if iteration > iterations:
+                break
+            img = imgs[x:(x + batch_size)]
+            if args.get('labels_numpy_path') is not None:
+                lb.append(lbs[x])
+            inputs = []
+            if args.get('dag-batch-size-auto'):
+                newShape = img.shape[0:1] + (1,) + img.shape[1:]
+                inputs.append(httpclient.InferInput(
+                    args['input_name'], newShape, "FP32"))
+            else:
+                inputs.append(httpclient.InferInput(
+                    args['input_name'], img.shape, "FP32"))
+            outputs = []
+            inputs[0].set_data_from_numpy(img)
+            data.append(triton_client.async_infer(
+                model_name=args.get(
+                    'pipeline_name') if is_pipeline_request else args.get('model_name'),
+                inputs=inputs,
+                outputs=outputs))
+
+            output = data[x].get_result().as_numpy(args['output_name'])

```

Comment:
Why for http there's a different approach than with grpc sample? This part calls async_infer method, but it looks like effectively it's synchronous as we call get_result() right after inference is scheduled.

Reply:
There doesn't seem to be a callback equivalent in the http client https://github.com/triton-inference-server/client/blob/1a98fe0cd6b18ded726ada052f1bab06ab07ceb1/src/python/library/tritonclient/http/__init__.py#L1494

Reply:
You're right. Let's stick to this version then.